Sprinkle H5_ATTR_FORMAT over printf(3)-like functions in tools and fix issues

gnuoyd opened the pull request.

gnuoyd committed:
Correct some conversion specifications.

gnuoyd committed:
Correct a format string.

gnuoyd committed:
Correct a format string.

gnuoyd committed:
Use the new convention, "%" PRIuHSIZE, instead of HSIZE_T_FORMAT.

gnuoyd committed:
Use the correct conversion specification for `long long` and `unsigned
long long`.

gnuoyd committed:
Replace "%" H5_PRINTF_LL_WIDTH "u" with "%" PRIuHSIZE where the argument
is `hsize_t`.

gnuoyd committed:
Label printf(3)-like functions with H5_ATTR_FORMAT(printf, m, n).

gnuoyd committed:
Remove `HSIZE_T_FORMAT` #definitions.

gnuoyd committed:
Remove an argument that's never formatted.

github-actions[bot] committed:
Committing clang-format changes

gnuoyd committed:
Replace many "%lld" occurrences with "%" PRIuHSIZE except for a few
instances where PRIdHSIZE was appropriate.  Remove a couple of casts and
use correct format strings, instead.

github-actions[bot] committed:
Committing clang-format changes

lrknox commented:
Checks passed in action https://github.com/HDFGroup/hdf5/actions/runs/1797059611.

byrnHDF commented on tools/src/h5dump/h5dump_xml.c:
```diff
@@ -4529,12 +4526,11 @@ xml_print_enum(hid_t type)
         else if (H5T_SGN_NONE == H5Tget_sign(native)) {
-            h5tools_str_append(&buffer, "%" H5_PRINTF_LL_WIDTH "u",
-                               *((unsigned long long *)((void *)(value + i * dst_size))));
+            /* XXX haven't I fixed this code somewhere else, before? */
```
XXX?

lrknox replied:
@gnuoyd is this comment to be kept, or was it a temporary note?

gnuoyd committed:
Copy values from a possibly unaligned buffer to aligned local variables
instead of casting and dereferencing pointers into the buffer.

github-actions[bot] committed:
Committing clang-format changes

gnuoyd commented:
On Fri, Feb 18, 2022 at 11:35:21AM -0800, Larry Knox wrote:
> @lrknox commented on this pull request.
> 
> 
> 
> > @@ -4529,12 +4526,11 @@ xml_print_enum(hid_t type)
>                  h5tools_str_append(&buffer, "%02x", value[i * dst_size + j]);
>          }
>          else if (H5T_SGN_NONE == H5Tget_sign(native)) {
> -            h5tools_str_append(&buffer, "%" H5_PRINTF_LL_WIDTH "u",
> -                               *((unsigned long long *)((void *)(value + i * dst_size))));
> +            /* XXX haven't I fixed this code somewhere else, before? */
> 
> @gnuoyd is this comment to be kept, or was it a temporary note?

I fixed the issues and removed the comments.

Dave

lrknox commented:
All checks passed in action https://github.com/HDFGroup/hdf5/actions/runs/1878179798
